File: 2.het/SzuloGyerek/Java/SzuloGyerek.java
class SzuloGyerek{

    public static void main (String[]args){
        Szulo szulo = new Gyerek();
        System.out.println("Gyerek letrehotasa: \n\tSzulo szulo = new Gyerek()");

        System.out.println("\nA szulo objektum fuggvenyet hasznaljuk:\n\tszulo.szulokiir()");

        szulo.szulokiir();

        System.out.println("\nA szulo.gyerekkiir()-re mar errort kapunk");

        szulo.gyerekkiir();

    }
}

File: 2.het/SzuloGyerek/Cpp/SzuloGyerek.cpp
#include <iostream>

class Szulo{
    public:
        void szulokiir(){
            std::cout << "Ez a szulo" << std::endl;
        } 
};

class Gyerek: public Szulo{
    public:
        void gyerekkiir(){
            std::cout << "Ez a gyerek" << std::endl;
        }
};

int main(){
    Szulo *szulo = new Gyerek();

    std::cout << "Gyerek létrehozása: \n\tSzulo *szulo = new Gyerek();" << std::endl;
    std::cout << "\nA szulo objektum szulokiir() fuggvenyet hasznaljuk: \n\tszulo->szulokiir()" << std::endl;
    szulo->szulokiir();
    std::cout << "\nA szulo objektum gyerekkiir()-re mar errort kapunk" << std::endl;
    szulo->gyerekkiir();
    delete szulo;  
}